Revamp Absurd

RobLoach opened the pull request:
- Brings in Boilerplate
- Brings in `test-jstransformer`
- Cleans up the processing method

TimothyGu committed:
Initial commit

TimothyGu committed:
Update README

TimothyGu committed:
Remove coveralls from devDependencies

Fixes #1.

RobLoach committed:
feat(inputFormats): Add inputFormats

This adds the inputFormats key to the boilerplate, allowing transformers to define what the expect input is:
https://github.com/jstransformers/jstransformer#inputformats

hemanth committed:
Merge pull request #3 from jstransformers/inputformats

feat(inputFormats): Add inputFormats

RobLoach committed:
docs(license): Switch LICENSE to markdown

TimothyGu committed:
Merge pull request #4 from jstransformers/license-md

Switch LICENSE to markdown

RobLoach committed:
fix(files): Add files to package.json

This will make npm only deploy index.js and LICENSE.md.

TimothyGu committed:
Merge pull request #5 from jstransformers/package-files

Add files to package.json

RobLoach committed:
Update and rename History.md to HISTORY.md

ForbesLindesay committed:
Merge pull request #8 from jstransformers/history

Update and rename History.md to HISTORY.md

RobLoach committed:
fix(files): Remove "LICENSE.md" from "files"

I believe it includes all `*.md` files automatically for us.

TimothyGu committed:
Merge pull request #9 from jstransformers/remove-license

Remove "LICENSE.md" from "files" in package.json

RobLoach committed:
Add some things

RobLoach committed:
Fix if statement wrapper, and add async functions

hemanth committed:
Merge pull request #11 from jstransformers/things

Add some things

RobLoach committed:
Add links in the documentation

RobLoach committed:
Disable email notifications

RobLoach committed:
Merge pull request #12 from jstransformers/datlinktho

Add links in the documentation

RobLoach committed:
Fix tests to trim the output

```
   Output:	"&lt;span class=&quot;foo&quot; id=&#39;bar&#39;&gt;hello world&lt;/span&gt;\n"
   Expected:	"&lt;span class=&quot;foo&quot; id=&#39;bar&#39;&gt;hello world&lt;/span&gt;"
 ✓ escape-html.render() (11ms)
```
While the tests pass, there's an extra \n at the end of output console log. Having the input to assertEqual() there will fix it.

RobLoach committed:
Send notification email only on failure

RobLoach committed:
Add compileFile() test

RobLoach committed:
Add compileFileAsync() test

RobLoach committed:
Merge pull request #17 from jstransformers/compileFile

Add compileFile() test

ForbesLindesay committed:
Merge pull request #13 from jstransformers/email

Disable email notifications

ForbesLindesay committed:
Merge pull request #15 from jstransformers/trim

Fix tests to trim the output

ForbesLindesay committed:
Merge pull request #18 from jstransformers/compileFileAsync

Add compileFileAsync() test

RobLoach committed:
Re-vamp Absurd with Boilerplate

Merge branch 'master' of github.com:jstransformers/boilerplate

Conflicts:
	.gitignore
	.travis.yml
	HISTORY.md
	LICENSE.md
	README.md
	index.js
	package.json
	test/index.js

RobLoach committed:
Fix additional merge conflicts

RobLoach committed:
Fix Travis

RobLoach committed:
Update documentation and clean dependencies

TimothyGu commented on index.js:
```diff
@@ -1,48 +1,63 @@
-    api = api(absurd, options);
+
+  // Process the input for the object.
+  if (typeof input == 'string' || input instanceof String) {
```
Use duck typing when possible.

Also, isn't `input` not allowed to be anything but a string?

RobLoach replied:
> Also, isn't input not allowed to be anything but a string?

Really not sure... Absurd has a lot of different inputs, and outputs.

RobLoach committed:
Update Boilerplate

Add different types for .render()

Fix whitespace

Fix whitespace

Fix whitespace

Fix to use test-jstransformer

Move tests up a directory

Add Dependencies badge

Update dependency badge to use DavidDM

We aren't using jscover

Add .editorconfig

Fix the copyright to reference the JSTransformers Team

Update License.md to JSTransformers Team

https://github.com/orgs/jstransformers/people

tunnckoCore commented:
:-1:

RobLoach commented:
> :-1: 

If you state where your disagreement lies, then we could fix things to satisfy your needs.

tunnckoCore commented:
https://github.com/jstransformers/jstransformer-gray-matter/pull/2#issuecomment-107745064
- will break my workflow, automation, sublime linter and fast fingers, :laughing:

RobLoach commented:
> will break my workflow, automation, sublime linter and fast fingers, 

Move additions to [Boilerplate](https://github.com/jstransformers/boilerplate), and they will be brought downstream to all the transformers.
